src/main/java/org/ballerinalang/containers/docker/impl/DefaultBallerinaDockerClient.java

```diff
@@ -152,7 +152,18 @@ public boolean deleteImage(String packageName, String dockerEnv, String imageNam
     public String getImage(String imageName, String dockerEnv) {
         DockerClient client = getDockerClient(dockerEnv);
         List<Image> images = client.image().list().filter(imageName).endImages();
+
+        // No images found?
+        if (images == null || images.size() < 1) {
+            return null;
+        }
+
         for (Image image : images) {
+            // Invalid tags found?
+            if (image.getRepoTags() == null) {
+                return null;
```

Comment:
this should be continue; right? in your earlier logic it was continue. with this it will return null even other images are valid.